File: source-code/Server/backend/src/main/java/com/iasdietideals24/backend/entities/OffertaTempoFisso.java
package com.iasdietideals24.backend.entities;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalTime;

@EqualsAndHashCode(callSuper = true)
@Getter
@Setter
@NoArgsConstructor
@Entity(name = "offerta_tempo_fisso")
public class OffertaTempoFisso extends OffertaDiCompratore {
    @ManyToOne(cascade = {CascadeType.MERGE, CascadeType.REFRESH})
    @JoinColumn(name = "asta_tempo_fisso_id_asta", nullable = false, foreignKey = @ForeignKey(name = "fk_asta_tempo_fisso_id_asta"))
    @NonNull
    private AstaTempoFisso astaRiferimento;

    // AllArgsConstructor
    public OffertaTempoFisso(@NonNull LocalDate dataInvio, @NonNull LocalTime oraInvio, @NonNull BigDecimal valore, @NonNull Compratore compratoreCollegato, @NonNull AstaTempoFisso astaRiferimento) {
        super(dataInvio, oraInvio, valore, compratoreCollegato);

        this.astaRiferimento = astaRiferimento;
        astaRiferimento.addOffertaRicevuta(this);
    }

    @Override
    public String toString() {
        return "OffertaTempoFisso(astaRiferimento=" + this.getAstaRiferimento().getIdAsta() + ") is a " + super.toString();
    }
}

File: CodiceSorgente/Client/DietiDeals24/app/src/main/java/com/iasdietideals24/dietideals24/model/offerta/OffertaTempoFisso.kt
package com.iasdietideals24.dietideals24.model.offerta

import com.iasdietideals24.dietideals24.model.account.AccountCompratore
import com.iasdietideals24.dietideals24.model.asta.AstaTempoFisso
import java.math.BigDecimal
import java.sql.Date
import java.sql.Time

class OffertaTempoFisso(
    dataInvio: Date,
    oraInvio: Time,
    valore: BigDecimal,
    compratoreCollegato: AccountCompratore,

    private var astaRiferimento: AstaTempoFisso
) : OffertaCompratore(dataInvio, oraInvio, valore, compratoreCollegato) {

    fun getAstaRiferimento(): AstaTempoFisso {
        return astaRiferimento
    }

    fun setAstaRiferimento(astaRiferimento: AstaTempoFisso) {
        this.astaRiferimento = astaRiferimento
    }
}